src/KM_Controls.pas

```diff
@@ -93,6 +93,7 @@   TKMControl = class
 
     fTimeOfLastClick: Cardinal; //Required to handle double-clicks
 
+    fOnChangeVisibility: TEvent;

```

Comment:
IMO events should only be used by external things, not by subclasses. I would do it the following way:
 - Add UpdateVisibility as a virtual procedure of TKMControl (doesn't need to be implemented in TKMControl, it's just for subclasses)
 - Call it in TKMControl.SetVisibility
 - In TKMPanel override UpdateVisibility to call UpdateVisibility on all children (as it does now)
 - In TKMDropCommon override UpdateVisibility to close the list if the control is not visible (as it does now)